src/Language/utils.php

```diff
@@ -3,23 +3,53 @@
 namespace Digia\GraphQL\Language;
 
 /**
- * @param int $cp
+ * Multi-byte compatible `chr`.
+ *
+ * Based on the Symfony's Mbstring polyfills.
+ *
+ * @param int $code
  * @return string
  */
-function chrUTF8(int $cp)
+function mbChr(int $code)
 {
-    return mb_convert_encoding(pack('N', $cp), 'UTF-8', 'UCS-4BE');
+    if (0x80 > $code %= 0x200000) {
+        return \chr($code);
+    }
+    if (0x800 > $code) {
+        return \chr(0xC0 | $code >> 6) . \chr(0x80 | $code & 0x3F);
+    }
+    if (0x10000 > $code) {
+        return \chr(0xE0 | $code >> 12) . \chr(0x80 | $code >> 6 & 0x3F) . \chr(0x80 | $code & 0x3F);
+    }
+
+    return \chr(0xF0 | $code >> 18) . \chr(0x80 | $code >> 12 & 0x3F) . \chr(0x80 | $code >> 6 & 0x3F) . \chr(0x80 | $code & 0x3F);
 }
 
 /**
+ * Multi-byte compatible `ord`.
+ *
+ * Based on the Symfony's Mbstring polyfills.
+ *
  * @param string $string
+ * @param string $encoding
  * @return int
  */
-function ordUTF8(string $string)
+function mbOrd(string $s)
```

Comment:
Decided to port the `chr` and `ord` functions from Symfony's Mbstring polyfill.

Reply:
Great. That should help a lot.